tox.ini

```diff
@@ -32,27 +36,37 @@ description =
     warnings: treating warnings as errors
     cov: with coverage
     xdist: using parallel processing
+allowlist_externals =
+    git
 pass_env =
     CRDS_*
     HOME
     CI
     TOXENV
     CODECOV_*
+set_env =
+    devdeps: UV_INDEX = https://pypi.anaconda.org/liberfa/simple https://pypi.anaconda.org/astropy/simple https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
+    devdeps: UV_INDEX_STRATEGY = unsafe-any-match
 extras =
     test
+uv_resolution =
+    oldestdeps: lowest-direct
 deps =
+    withromancal: romancal @ git+https://github.com/spacetelescope/romancal.git
     xdist: pytest-xdist
-    cov: pytest-cov
-    oldestdeps: minimum_dependencies
+    cov: pytest-cov >= 4.1.0
     rad: rad @ git+https://github.com/spacetelescope/rad.git@main#egg=rad
+    romancal: romancal[test] @ git+{[main]rcal_repo}
+    devdeps: -r requirements-dev.txt
+change_dir =
+    romancal: {env_tmp_dir}
 commands_pre =
-    oldestdeps: minimum_dependencies roman_datamodels --filename requirements-min.txt
-    oldestdeps: pip install -r requirements-min.txt
-    devdeps: pip install -r requirements-dev.txt -U --upgrade-strategy eager
-    pip freeze
+    {list_dependencies_command}
+    git clone {[main]rcal_repo}
```

Comment:
Should be able to use `git archive --remote={[main]rcal_repo} main pyproject.toml | tar -d` but I couldn't get `git archive` to connect for some reason.

Reply:
GitHub apparently does not support archive over https.

Used a `git clone -n --depth=1 --filter=blob:none {[main]rcal_repo}` instead this should only get the `.git` without any blob files for the last commit. I can then checkout the single file I care about.